src/backend/distributed/commands/dependencies.c

```diff
@@ -245,6 +248,12 @@ ReplicateAllDependenciesToNode(const char *nodeName, int nodePort)
 		ObjectAddress *dependency = (ObjectAddress *) lfirst(dependencyCell);
 		ddlCommands = list_concat(ddlCommands,
 								  GetDependencyCreateDDLCommands(dependency));
+
+		if (!*clusterHasDistributedFunction &&
```

Comment:
This API is kind of weird. Maybe just add a separate function that looks whether there is any tuple with `ProcedureRelationId` in the `classid` column in `pg_dist_object`.

Reply:
I didn't want to scan through the `pg_dist_object` again, but that's OK as well.